docs/apache-airflow/concepts/dags.rst

```diff
@@ -337,7 +337,7 @@ However, this is just the default behaviour, and you can control it using the ``
 * ``none_failed``: All upstream tasks have not ``failed`` or ``upstream_failed`` - that is, all upstream tasks have succeeded or been skipped
 * ``none_failed_or_skipped``: All upstream tasks have not ``failed`` or ``upstream_failed``, and at least one upstream task has succeeded.
 * ``none_skipped``: No upstream task is in a ``skipped`` state - that is, all upstream tasks are in a ``success``, ``failed``, or ``upstream_failed`` state
-* ``dummy``: No dependencies at all, run this task at any time
+* ``null``: No dependencies at all, run this task at any time
```

Comment:
Given this definition, is `null` the best name? It doesn't feel like a perfect fit to me, but I'm also racking my brain to come up with something better. I wonder what others think. Some options off the top of my head:
1. `instant`
1. `unrestricted`
1. `immediate`

Reply:
Maybe `standalone`?

Reply:
I took the name from the discussion in https://github.com/apache/airflow/issues/15994

Reply:
Just a thought .. since we are in Python world maybe `None` operator is a better world (it also fits a bit better when it comes to the meaning) ?

Reply:
I'm happy with any of the names suggested.
We had `dummy` trigger rule and `DummyOperator` but we don't have to choose the same name for both we can have `None` trigger rule and `NullOperator`.

I can do the change now or wait for more thoughts.. I assume there will be more opinions on that issue.

Reply:
Similar to the `None` state we have for DagRun and TaskInstances, maybe we just have `None` as a valid option instead of 'dummy'

Reply:
Personally I prefer a special null/none trigger rule instead of using Python’s None directly. None makes every type annotation more difficult (`Optional` everywhere) and causes bugs more easily (None is used by a lot of code as the sentinel). The latter is not as significant for Airflow since `TriggerRule` values are plain strings anyway, but the former point still stands; `Optional[str]` everywhere is cumbersome in many ways.

Reply:
+1 on `None` for the trigger rule. (No real preference on `None` vs `"None"`)

Reply:
I like `None`/`"None"` better than `null` for sure.

But just to hammer my original point home a bit more: the other names are much more explicit and grammatical, e.g. *"trigger when all_succeeded"* or *"trigger when one_failed"*. From reading the name it's very clear what the behaviour is. But *"trigger when None"* feels a bit out of place and quite vague. To me something like *"trigger immediately"* or *"trigger when now"* are even more descriptive of how the trigger actually behaves and matches the other trigger names more closely.

But if I'm still out voted I think `None` is a significant improvement :)